File: Day25/DSA/Find_Missing_and_Repeated_Values.cpp
Question:-  https://leetcode.com/problems/find-missing-and-repeated-values/description/

Code:- 

class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        int sum = 0;
        int orisum = 0;
        unordered_map<int, int> mp;
        for (int i = 0; i < grid.size(); i++) {
            for (int j = 0; j < grid[i].size(); j++) {
                mp[grid[i][j]]++;
                sum += grid[i][j];
            }
        }
        //  cout<<sum<<endl;;

        vector<int> v;
        for (auto i : mp) {
            if (i.second == 2)
                v.push_back(i.first);
        }

        // for (int i = ; i < grid.size() * grid.size(); i++) {
        //     orisum += i;
        // }

        int n = grid.size()*grid.size();

        orisum = ((n*n) + n)/2;

        // cout<<orisum;

        v.push_back(orisum - sum + v[0]);
        return v;
    }
};

File: Day25/DSA/Find_Missing_and_Repeated_Values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {

        int sum = 0;
        int orisum = 0;

        HashMap<Integer, Integer> mp = new HashMap<>();

        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[i].length; j++) {
                mp.put(grid[i][j], mp.getOrDefault(grid[i][j], 0) + 1);
                sum += grid[i][j];
            }
        }

        ArrayList<Integer> list = new ArrayList<>();

        for (Map.Entry<Integer, Integer> entry : mp.entrySet()) {
            if (entry.getValue() == 2) {
                list.add(entry.getKey());
            }
        }

        int n = grid.length * grid.length;

        orisum = (n * (n + 1)) / 2;

        list.add(orisum - sum + list.get(0));

        int[] ans = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            ans[i] = list.get(i);
        }

        return ans;
    }
}
